src/table/hooks/use-selection-reducer.ts

```diff
@@ -18,8 +19,11 @@ const useSelectionReducer = (
     isSelectMultiValues: false,
   });
 
+  // Only update pageRows for pagination table from here, VT is doing it in a different hook
   useDidUpdateEffect(() => {
-    selectionDispatch({ type: SelectionActions.UPDATE_PAGE_ROWS, payload: { pageRows } });
+    if (isPagination) {
```

Comment:
Was it not possible to move this hook call into `TableWrapper`? Would not need the `isPagination` flag than

Reply:
never tried, actually. worth giving a try

Reply:
I think we get an extra render, but it 10 vs 11 so it doesn't really matter. It was for confirming selections, so not that strange that a lot of things happen. Performance is not that important when rows updates